create/create.go

```diff
@@ -30,59 +28,134 @@ type Cmd struct {
 type resultFunc func(watch.Event) (bool, error)
 
 type creator struct {
-	kind       string
-	mg         resource.Managed
-	objectList runtimeclient.ObjectList
+	client *api.Client
+	mg     resource.Managed
+	kind   string
 }
 
-func newCreator(mg resource.Managed, resourceName string, objectList runtimeclient.ObjectList) *creator {
-	return &creator{mg: mg, kind: resourceName, objectList: objectList}
+type waitStage struct {
+	kind        string
+	waitMessage *message
+	doneMessage *message
+	objectList  runtimeclient.ObjectList
+	listOpts    []runtimeclient.ListOption
+	onResult    resultFunc
 }
 
-func (c *creator) createResource(ctx context.Context, client *api.Client) error {
-	if err := client.Create(ctx, c.mg); err != nil {
+type message struct {
+	icon     string
+	text     string
+	disabled bool
+}
+
+func (m *message) progress() string {
+	if m.disabled {
+		return ""
+	}
+
+	return format.ProgressMessagef(m.icon, m.text)
+}
+
+func (m *message) printSuccess() {
+	if m.disabled {
+		return
+	}
+
+	format.PrintSuccessf(m.icon, m.text)
+}
+
+func newCreator(client *api.Client, mg resource.Managed, resourceName string) *creator {
```

Comment:
is this actually needed? Seems to just pass function args to a struct.

Reply:
It's not really needed right now but if we add new fields to `creator` at some point they could be defaulted here without needing to change all call-sites. Also this makes it possible to do something like `newCreator(...).wait(...)`, which you can't do when creating a new struct.

Reply:
Ok, lets leave it in then. I was curious about the last statement. Not sure if I did it correctly, but it seems to work for me: https://go.dev/play/p/7sjk41nKy8C.

Reply:
Ah right, forgot that you can wrap the struct initialization in braces, looks pretty ugly though 😄